src/stdlib/match_datadog_query.rs

```diff
@@ -120,7 +120,13 @@ impl Resolver for VrlFilter {}
 /// Implements `Filter`, which provides methods for matching against (in this case) VRL values.
 impl Filter<Value> for VrlFilter {
     fn exists(&self, field: Field) -> Box<dyn Matcher<Value>> {
-        let buf = lookup_field(&field);
+        let buf = match lookup_field(&field) {
+            Ok(path) => path,
+            Err(error) => {
+                error!(message = "Invalid query field", %error);
+                return Run::boxed(|_| false);
```

Comment:
I was thinking we'd change this function to return an error instead of logging it directly. Also, will this log for every event flowing through the system? It could flood, if so, in which case we should add `internal_log_rate_limit = true`.

Reply:
Ah, I see your note about further refactoring. I'd be ok with just adding `internal_log_rate_limit = true` now if this would flood otherwise and refactor the function signature in a follow-up.

Reply:
Gotcha, I realize that it doesn't make sense to follow up in a second PR. I will update this PR.

Reply:
This commit https://github.com/vectordotdev/vrl/pull/1031/commits/551b0d21bc9521df08ef546fd557d7c3dc526573 eliminated all `error!` calls.